### ISSUE TITLE
[feature] Ability to screen/approve instances before fully federating with them

### ISSUE THREAD

**Issue by goatrocks** [superseriousbusiness/gotosocial#515]
I would like to propose a fourth federation mode in addition to the three described in the roadmap, where newly discovered instances are queued for approval like new accounts are. This would serve to pre-screen new instances before they get full permission to interact with an instance. This would be a nice middle ground between allow list and full federation.

**Comment by tsmethurst** [comment]
Agreed! Good suggestion. We'll have to think a bit more about how exactly to do this, since an instance needs to interact with your instance in *some* way before your instance becomes aware of it. Perhaps a dereference request or a like would be enough to put the instance in the queue for approval?

**Comment by OfficialBoyfriend** [comment]
The current federation modes seems to be sufficient. Excessive enhancement of censorship/filtering related functions seems to be detrimental to the development of the federation universe, On the contrary, it is easy to cause the division of the federation universe. No one likes a censorship-laden community, and implementing relevant features makes censorship fairly easy.

Of course, this is just my personal opinion, nothing to worry about. From Google Translate.

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/515.
